src/App.js

```diff
@@ -1,16 +1,68 @@
 import Welcome from "./pages/Welcome";
 import Timeline from "./pages/Timeline";
+import "./App.css";
 import { Route, Routes } from "react-router-dom";
 import NavBar from "./components/NavBar/NavBar";
-import "./App.css";
+import { useAuth0 } from "@auth0/auth0-react";
+import axios from "axios";
+import { useEffect } from "react";
+import Register from "./pages/Register";
+import { ProtectedRoute } from "./components/Auth/ProtectedRoute";
+import PageLoader from "./pages/PageLoader";
+import Messages from "./pages/Messages";
+import Profile from "./pages/Profile";
+import { useUserContext } from "./context/UserContext";
 
 function App() {
+  const { isLoading, isAuthenticated, getAccessTokenSilently } = useAuth0();
+  const { user, setUser } = useUserContext();
+
+  useEffect(() => {
+    const getUser = async () => {
+      try {
+        const token = await getAccessTokenSilently();
+        const response = await axios.get("http://localhost:3001/api/users/", {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        });
+        setUser(response.data);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    getUser();
+  }, [getAccessTokenSilently, setUser]);
+
+  if (isLoading || (isAuthenticated && user.isLoading)) {
```

Comment:
We could maybe take these three if statements and create an if-else statement to make it easier to look at, but there would not be an else option, so I don't really see a point in that and we can just leave it as it

Reply:
If I remember correctly you can't use if-else for returning react components. It has to be an if